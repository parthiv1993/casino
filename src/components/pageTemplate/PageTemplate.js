import { Outlet } from 'react-router-dom';
import logo from '../../assets/logo.svg';
import { LogoImage, PageContent, PageWrapper } from './PageTemplate.styles';

const PageTemplate = () => {
  return (
    <PageWrapper>
      <LogoImage src={logo} alt="logo" />
      <PageContent>
        <Outlet />
      </PageContent>
    </PageWrapper>
  );
};

export default PageTemplate;
