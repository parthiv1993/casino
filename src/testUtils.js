import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import light from './contexts/theme/light';

export const renderComponentWithLightTheme = (children) => {
  return render(<ThemeProvider theme={light}>{children}</ThemeProvider>);
};

export const lowerCaseAllKeys = (props = {}) => {
  const lowercaseObject = {};
  Object.keys(props).map((key) => (lowercaseObject[key.toLowerCase()] = props[key]));

  return lowercaseObject;
};

export const MockIcon = (props) => <div component="Icon" {...lowerCaseAllKeys(props)} />;
