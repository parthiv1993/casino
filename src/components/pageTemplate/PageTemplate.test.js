import PageTemplate from './PageTemplate';
import { renderComponentWithLightTheme } from '../../testUtils';

describe('PageTemplate ', () => {
  it('should match snapshot', () => {
    const { container } = renderComponentWithLightTheme(<PageTemplate />);
    expect(container).toMatchSnapshot();
  });
});
