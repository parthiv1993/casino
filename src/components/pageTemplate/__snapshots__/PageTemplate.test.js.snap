// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTemplate  should match snapshot 1`] = `
.c0 {
  background-color: #444;
  width: 100vw;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  width: 100%;
  margin-inline: auto;
  background-color: white;
}

@media (max-width:425px) {
  .c1 {
    width: 300px;
    height: 60px;
  }
}

<div>
  <div
    class="c0"
  >
    <img
      alt="logo"
      class="c1"
      height="135"
      src="logo.svg"
      width="720"
    />
    <section
      class="c2"
    />
  </div>
</div>
`;
